### ISSUE TITLE
[BUG] Alignment issue with CBLC/EBLC index format 3

### ISSUE THREAD

**Issue by ckitagawa-work** [fonttools/fonttools#1817]
According to the spec CBLC/EBLC IndexSubtables using index format 3 need to be 32-bit aligned [1]. However, due to a bug in calculating if padding is required near line 482 of `E_B_L_C_.py` the resulting compiled table will always be 16-bit aligned. See the code for an explanation.

```python
dataList = [EblcIndexSubTable.compile(self, ttFont)]
# At this point len(dataList) == 1 this will be a problem when doing padding calculation.
# The first entry in dataList is already 32-bit aligned as it is:
# uint16 firstGlyphId
# uint16 lastGlyphId
# uint32 imageDataOffset

dataList += [struct.pack(dataFormat, offsetValue) for offsetValue in offsetArray]
# Now len(dataList) == 1 + # of offsets

# Only # of offsets should be considered for padding not the first entry in dataList
# len(dataList) should be len(offsetArray) or len(dataList) - 1
if offsetDataSize * len(dataList) % 4 != 0:
    dataList.append(struct.pack(dataFormat, 0))
```

[1] [Spec for IndexSubtables](https://docs.microsoft.com/en-us/typography/opentype/spec/eblc#indexsubtables)

**Comment by anthrotype** [comment]
ouch! thanks for spotting this. I'll replace:

```diff
- if offsetDataSize * len(dataList) % 4 != 0:
+ if offsetDataSize * len(offsetArray) % 4 != 0:
```

I wonder how many fonts out there are affected by this. Did you find any in the wild yet?

**Comment by anthrotype** [comment]
NotoColorEmoji.ttf seems to only contain `eblc_index_sub_table_1`. Do you happen to have a test font at hand so we can add a test for this?

**Comment by ckitagawa-work** [comment]
I'm working on adding subsetting support for CBLC/CBDT to Harfbuzz and found this bug while adding/debugging some tests. Unfortunately, I don't have any real in-the-wild examples. I just modified NotoColorEmoji.ttx to use format 3 and recompiled to a ttf file. I attached the TTX file and I used harfbuzz to make a valid ttf version...

[NotoColorEmoji.index_format_3.zip](https://github.com/fonttools/fonttools/files/4140719/NotoColorEmoji.index_format_3.zip)
